ext/mbstring/php_mbregex.c

```diff
@@ -897,15 +899,21 @@ static int _php_mb_onig_search(regex_t* reg, const OnigUChar* str, const OnigUCh
 static void _php_mb_regex_ereg_exec(INTERNAL_FUNCTION_PARAMETERS, int icase)
 {
 	zval *array = NULL;
-	char *arg_pattern, *string;
-	size_t arg_pattern_len, string_len;
+	char *string;
+	size_t string_len;
+	zend_string *pattern;
```

Comment:
Why this change? This would make sense to me if `php_mbregex_compile_pattern()` was converted to accept zend_string and all callers changed to pass zend_string, but as-is, this makes this this code the odd duck out.

Reply:
So I think my rationale was to work backwards but then I realized that it uses a lot of the str* and str_len* stuff so I'm not sure how if it would be better to replace those with the ZSTR_VAL(pattern) and ZSTR_LEN(pattern) ... could always do that and see how it looks
(yes I'm rambling)